PAPER.md

```diff
@@ -0,0 +1,378 @@
++                                                               +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                        message hash(256)                      |
++       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|       |                                                       |
++-+-+-+-+                                                       +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                     acknowledger(256)                         |
++       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|       |        signature len(16)      |                       |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+                       +
+|                            signature                          |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+```
+
+* `type` field represents an unsigned integer with value `5`
+
+* `uuid` field represents the identifier of the message we are acknowledging.
+
+* `message hash` field represents the SHA256 sum of the message its being acknowledged.
+
+* `acknowledger` represents acknowledger `sha256` public `key fingerprint`.
+
+
+Receivers of the message acknowledge will save this acknowledges in disk. They are interested on that.
+
+### Message reply and acceptance
+
+Message reply refers to the ability for an user to mention, or reply a specific message. This covers scenarios in which we want a proof
+of acceptance regarding a specific message. Lets see an example:
+
+1. Alice: Hey Bob ! i am thinking on selling my car. Would you accept it for 10k $ ?
+2. Alice: Please think about it :)
+3. Bob: Hey Alice, looks like a good deal. I accept it.
+
+In the above example, Bob its interested in accepting the offer. In order to do that, instructs the application to 
+"reply this message" for message `1`. Under the hood, the application sets 2 special headers (As specified in the message section) in the reply message (`3`) : 
```

Comment:
Probably lacking context here, but `Message reply and acceptance` seems like a core feature that shouldn't rely on "special headers" to work properly. The main reason against this is that `headers` is just a dynamic field that won't be versioned according to the protocol.

So my question here is, why not using a dedicated field? Are we able to reply and accept if we don't provide this headers?

Reply:
I had similar thoughts :smile: and finally decided for adding them as headers. Here is my rationale:

I think the headers are exclusively used by the protocol/application. Users are not able to add headers to the message. Its still internal machinery of the protocol.  Adding this 2 fields to the main `message` would result in precious wasted bits and added complexity most of the times, as the "reply" feature its not always used. If there is a change in any header in newer versions, that should be properly documented and we should release a major version (breaking change).

I am thinking that _maybe_ the name `headers` its confusing. Maybe `attributes` its a better one and we should add a clarification about this. Let me know your thoughts.